src/lib/DocSearch.js

```diff
@@ -154,6 +154,21 @@ class DocSearch {
     });
   }
 
+  static formatURL(hit) {
+    const {url, anchor} = hit;
+    if (url) {
+      const containsAnchor = url.indexOf('#') !== -1;
+      if (containsAnchor) return url;
+      else if (anchor) return `${hit.url}#${hit.anchor}`;
+      return url;
+    }
+    else if (anchor) return `#${hit.anchor}`;
```

Comment:
When would that happen? If we end up with a record with no `url` but an `anchor` field, then we have an issue with the crawler :)